src/Lean/PrettyPrinter/Delaborator/Basic.lean

```diff
@@ -216,22 +219,46 @@ where
  }
 
 def addFieldInfo (pos : Pos) (projName fieldName : Name) (stx : Syntax) (val : Expr) : DelabM Unit := do
-  let info ← mkFieldInfo projName fieldName stx val
+  let info := Info.ofFieldInfo <| ← mkFieldInfo projName fieldName stx val
   modify fun s => { s with infos := s.infos.insert pos info }
 where
-  mkFieldInfo projName fieldName stx val := return Info.ofFieldInfo {
+  mkFieldInfo projName fieldName stx val := return {
     projName := projName,
     fieldName := fieldName,
     lctx := (← getLCtx),
     val := val,
     stx := stx
   }
 
+def addOmissionInfo (pos : Pos) (stx : Syntax) (e : Expr) : DelabM Unit := do
+  let info := Info.ofOmissionInfo <| ← mkOmissionInfo stx e
+  modify fun s => { s with infos := s.infos.insert pos info }
+where
+  mkOmissionInfo stx e := return {
+    toTermInfo := ← addTermInfo.mkTermInfo stx e (isBinder := false)
+  }
+
+def withIncDepth (act : DelabM α) : DelabM α := fun ctx =>
+  act { ctx with depth := ctx.depth + 1 }
+
+def isMaxDepthReached : DelabM Bool := do
+  let maxDepthCheckEnabled := ! (← getPPOption getPPDeepTerms)
+  let maxDepth ← getPPOption getPPMaxTermDepth
+  let depth := (← read).depth
+
+  return maxDepthCheckEnabled && depth > maxDepth
```

Comment:
This function is reading both pretty printer options every time. I would suggest modifying it in the following way (assuming we switch to `pp.omitDeepTerms`), and also adding a docstring.
```lean
/--
Returns `true` if the delaborator should omit a term and replace it with `⋯`.
-/
def isMaxDepthReached : DelabM Bool := do
  unless (← getPPOption getPPOmitDeepTerms) do return false
  let maxDepth ← getPPOption getPPMaxTermDepth
  let depth := (← read).depth
  return depth > maxDepth
```
(I also wouldn't find it any less readable if the last two lines were coalesced into `return (← read).depth > maxDepth`, but that's a very minor suggestion.)